src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -213,6 +215,44 @@ protected final String getUriString(String filename) {
                 .toString();
     }
 
+    /**
+     * Returns the absolute system path of the given file.
+     *
+     * @param filename file name.
+     * @return absolute file path.
+     */
+    protected final File getInputFile(String filename) {
+        return new File("src/test/resources/" + getPackageLocation() + "/" + filename);
+    }
+
+    /**
+     * Performs verification of the config read from input file.
+     *
+     * @param testConfig hardcoded test config.
+     * @param parsedConfig parsed config from input file.
+     */
+    protected static void verifyConfig(DefaultConfiguration testConfig,
+                                      DefaultConfiguration parsedConfig) {
+        final String errorMessage = "Test configs and parsed configs differ.";
+        assertWithMessage(errorMessage)
+                .that(testConfig.getName())
+                .contains(parsedConfig.getName());
```

Comment:
why not `isEqualTo` ?

Reply:
if you wanted to cover `MissingOverrideCheck` and `MissingOverride`, you can just add `Check` suffix explicitly
or double compare for `parsedConfig.getName()` and `parsedConfig.getName() + Check`

but I think we can add `Check` for now

Reply:
Nope!
The test methods instantiate the Config using ```MissingOverrideCheck.class```
The Bdd parses instantiates it using ```"MissingOverride"```
Internally, ```MissingOverrideCheck.class``` maps to the entire package name ```com.puppycrawl.tools.checkstyle.checks.annotation.MissingOverrideCheck```

Truth says:
```
Test configs and parsed configs differ.
expected: MissingOverride
but was : com.puppycrawl.tools.checkstyle.checks.annotation.MissingOverrideCheck
Expected :MissingOverride
Actual   :com.puppycrawl.tools.checkstyle.checks.annotation.MissingOverrideCheck
<Click to see difference>
```

Reply:
hmm

We have future plans to migrate to BDD, it implies that we will replace `new DefaultConfiguration(clazz.getName());` with yours `new DefaultConfiguration(checkName);`

I suppose they should be identical, no?

Reply:
one approach is to update `AbstractCheckstyleModuleTestSupport.createModuleConfig` with `Class.getSimpleName` and see that no violation will occur

Reply:
another one is to create full qualified name during check name parsing

Reply:
> I suppose they should be identical, no?

```new DefaultConfiguration("MissingOverride")```
In this case, we need to provide ```ModuleCreationOption.IN_TREEWALKER``` as parent when creating checker.

Reply:
because this
https://github.com/checkstyle/checkstyle/blob/b1d2ebf2f801d14e5c3a5da8b0e26e624db9c161/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L103-L122
malfunctions when executing ```final Class<?> moduleClass = Class.forName(moduleName);```
```Class.forName()``` requires full package name.

Reply:
> In this case, we need to provide ModuleCreationOption.IN_TREEWALKER as parent when creating checker.

okey, lets do this way

Reply:
hold on, not done
we need to agree how to create DefaultConfiguration and it should be consistent with current DefaultConfiguration instantiation

Reply:
@romani FYI

Reply:
> hold on, not done
> we need to agree how to create DefaultConfiguration and it should be consistent with current DefaultConfiguration instantiation

Right. The DefaultCofiguration shall be the same, just that we'll need to supply ```ModuleCreationOption.IN_TREEWALKER``` or ```ModuleCreationOption.IN_CHECKER``` as parent while creating Checker.
Right now we are using the test config for checking so it's not a problem. We will do this when we use the parsed config for checking. Do you want it in this PR?

Reply:
maybe we can use package path from the input file? it should be the same as for check

Reply:
> maybe we can use package path from the input file? it should be the same as for check

This can be done.

Reply:
@shashwatj07 please do

Reply:
done